indigo_api/static/xsl/fo/_hier.xsl

```diff
@@ -0,0 +1,239 @@
+          </xsl:variable>
+          <xsl:choose>
+            <!-- certain containers get their heading on the next line -->
+            <xsl:when test="self::akn:book or self::akn:chapter or self::akn:title or self::akn:tome">
+              <fo:block keep-with-previous="always">
+                <xsl:apply-templates select="akn:heading"/>
+              </fo:block>
+            </xsl:when>
+            <!-- certain containers get a dash before the heading, unless there is a num and it ends in a stop or colon -->
+            <xsl:when test="self::akn:article or self::akn:clause or self::akn:part">
+              <xsl:if test="not($terminus='.' or $terminus=':')">
+                <xsl:text> –</xsl:text>
+              </xsl:if>
+              <xsl:text> </xsl:text>
+              <xsl:apply-templates select="akn:heading"/>
+            </xsl:when>
+            <!-- all other containers get a dash before the heading, only if there is a num and it doesn't end in a stop or colon -->
+            <xsl:otherwise>
+              <xsl:if test="akn:num and not($terminus='.' or $terminus=':')">
+                <xsl:text> –</xsl:text>
+              </xsl:if>
+              <xsl:text> </xsl:text>
+              <xsl:apply-templates select="akn:heading"/>
+            </xsl:otherwise>
+          </xsl:choose>
+        </xsl:if>
+        <xsl:if test="akn:subheading">
+          <fo:block font-weight="bold" font-size="{$fontsize-h3}">
+            <xsl:apply-templates select="akn:subheading"/>
+          </fo:block>
+        </xsl:if>
+      </fo:block>
+      <fo:block margin-top="{$para-spacing}">
+        <xsl:apply-templates select="./*[not(self::akn:num|self::akn:heading|self::akn:subheading)]"/>
+      </fo:block>
+    </fo:block-container>
+  </xsl:template>
+
+  <!-- schedules
+   - heading centered
+   - subheading centered
+   - content
+   -->
+  <xsl:template match="akn:attachment">
+    <fo:block margin-top="{$para-spacing}*2" font-size="{$fontsize-h2}" text-align="center" widows="2" orphans="2" keep-with-next="always" id="{@eId}">
+      <fo:inline font-weight="bold">
+        <xsl:apply-templates select="akn:heading"/>
+      </fo:inline>
+    </fo:block>
+    <fo:block margin-top="{$para-spacing}" font-size="{$fontsize-h2}" text-align="center" widows="2" orphans="2" keep-with-next="always">
+      <fo:inline font-weight="bold">
+        <xsl:apply-templates select="akn:subheading"/>
+      </fo:inline>
+    </fo:block>
+    <xsl:apply-templates select="./*[not(self::akn:heading|self::akn:subheading)]"/>
+  </xsl:template>
+
+  <!-- basic unit
+   - startQuote if it's the first element in the quote
+   - number to the side, bold (if present)
+   - heading in bold (if present)
+   - subheading in bold (if present)
+   - content in next block regardless of whether there's a heading -->
+  <xsl:template match="akn:rule|akn:section">
+    <!-- use a block container to retain relative indentation (nesting depth) -->
+    <fo:block-container>
```

Comment:
Idea for later: It might be useful to have all of this as a `basic-unit` template, and then the template for `akn:rule` etc. just calls that template. It would make it easier to add other things as a basic unit (but this could also be unnecessary future-proofing).